### PR TITLE
dialects / bug : (linalg) linalg.generic constructor does not have a result types argument

### LINKED ISSUE #2115: bug: linalg.generic constructor does not have a result types argument

### PR DESCRIPTION
**Pull request by kayode-gif**
resolves #2115, where no argument is present for results type. 
Note: Parsing code for the result is also added in this PR

### CODE REVIEW

**Comment by kayode-gif** [comment]
@superlopuh  realised the res stuff needs to be parsed too aswell (inside the parse methode), doing this in the same PR okay?

**Comment by superlopuh** [comment]
Yep at this point let's do the same PR, also can you copy the filecheck test for the linalg operation with a result to parse and print?

File: tests/filecheck/mlir-conversion/with-mlir/dialects/linalg/ops.mlir [hunk 1]
```diff
@@ -13,6 +13,16 @@ linalg.generic {indexing_maps = [affine_map<(d0, d1) -> ()>, affine_map<(d0, d1)
     linalg.yield %arg3 : f32
 }
 
+%2, %3 = "test.op"() : () -> (tensor<2x3xf32>, tensor<2x3xf32>)
+
+%sum = linalg.add ins(%2, %2 : tensor<2x3xf32>, tensor<2x3xf32>) outs(%3 : tensor<2x3xf32>) -> tensor<2x3xf32>
```
Comment (superlopuh) [inline tests/filecheck/mlir-conversion/with-mlir/dialects/linalg/ops.mlir]: this line should probably go into the PR that adds linalg.add

File: tests/filecheck/mlir-conversion/with-mlir/dialects/linalg/ops.mlir [hunk 2]
```diff
@@ -26,3 +34,10 @@ linalg.generic {indexing_maps = [affine_map<(d0, d1) -> ()>, affine_map<(d0, d1)
 // CHECK-NEXT:      linalg.yield %in : f32
 // CHECK-NEXT:    }
 // CHECK-NEXT:  }
+// CHECK-NEXT:    %1:2 = "test.op"() : () -> (tensor<2x3xf32>, tensor<2x3xf32>)
+// CHECK-NEXT:    %2 = linalg.generic {indexing_maps = [#map1, #map1, #map1], iterator_types = ["parallel", "parallel"]} ins(%1#0, %1#0 : tensor<2x3xf32>, tensor<2x3xf32>) outs(%1#0 : tensor<2x3xf32>) {
```
Comment (superlopuh) [inline tests/filecheck/mlir-conversion/with-mlir/dialects/linalg/ops.mlir:38]:
```suggestion
// CHECK-NEXT:    %{{.*}} = linalg.generic {indexing_maps = [#map1, #map1, #map1], iterator_types = ["parallel", "parallel"]} ins(%1#0, %1#0 : tensor<2x3xf32>, tensor<2x3xf32>) outs(%1#0 : tensor<2x3xf32>) {
```
etc

Comment (superlopuh) [reply tests/filecheck/mlir-conversion/with-mlir/dialects/linalg/ops.mlir:38]: I can try to reproduce locally if you'd like

**Review by superlopuh — APPROVED** [review]

**Review by compor — APPROVED** [review]

**Pull request merged**